### PR TITLE
Fixes #7536: Make possible to delete ANY active technique

### PR DESCRIPTION
**Pull request by fanf**
https://www.rudder-project.org/redmine/issues/7536

### CODE REVIEW

File: rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala [hunk 1]
```diff
@@ -210,7 +212,9 @@ class TechniqueEditForm(
         case e:EmptyBox => NodeSeq.Empty
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala]: There should be a log here or at lower level if the request failed (if it had to be launched)

Comment (fanf) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala]: I'm not sure, the code seems to use the box as an Option. I still can log on failure.

Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala]: You are right, it use empty as "there is no active technique", so don't log on emptybox but only on failure

Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala]: It's just that it's impossible to know that it fails

**Comment by fanf** [comment]
pr updated

File: rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala [hunk 2]
```diff
@@ -147,9 +149,18 @@ class TechniqueEditForm(
 //  }
 
 
-  private[this] var currentActiveTechnique = roActiveTechniqueRepository.getActiveTechnique(technique.id.name)
+  private[this] var currentActiveTechnique: Box[ActiveTechnique] =  Box(activeTechnique).or(Box(technique).flatMap(t => roActiveTechniqueRepository.getActiveTechnique(t.id.name)))
   private[this] var uptCurrentStatusIsActivated = currentActiveTechnique.map( _.isEnabled)
 
+  currentActiveTechnique match {
+    case f: Failure =>
+      logger.warn(s"An error was encoutered when trying to find a technique in user library: ${f.messageChain}")
```
Comment (VinceMacBuche) [inline rudder-web/src/main/scala/com/normation/rudder/web/components/TechniqueEditForm.scala]: encountered :)

**Comment by fanf** [comment]
PR updated

**Pull request merged**